### ISSUE TITLE
v156 admin relogin after timeout

### ISSUE THREAD

**Issue by mc12345678** [zencart/zencart#1630]
Fresh install as of 25 Feb 2018 about 0600 GMT.

If working in the admin and enough time has lapsed to effectively be logged out, the option to login again provides the following message: "`Your session has timed out. You were inactive, so we logged you out automatically. Please login to resume your session.`"

However, so far I have found that the user is returned to the index.php page rather than even the same page that was being edited at the time.  I didn't really see that this issue was a duplicate of any other, but if so please appropriately handle.

**Comment by drbyte** [comment]
Is your point that `Please login to resume your session.` implies they'd be taken immediately back to the spot they were at before the timeout? 

If so, then the wording is misinterpreted. It's nearly impossible to take them back to where they were if the session has fully expired, since there's no remaining "state" to use to determine the last activity.

**Comment by mc12345678** [comment]
> Consider the case of online banking: if you're logged out due to inactivity, when you login again you simply start back at the same place you'd arrive after any other login, ie: home page.

If this is/was part of what is required to maintain PCI compliance then I'm okay with it; however, I find I would rather disable the current final "relogin" than keep it, because if I were to simply click a menu option of where next to be upon relogin or reload the current page I would again at least have some control of it knowing that I picked something specific. (yes I can still "reload" the current page even with this message, I'm just reporting what was seen and my "initial" thoughts.) I think that the current page should be stored upon page load so that the popup javascript will try (if such is possible for that page) to reload the page. Otherwise, if the expectation is to return back to the login screen, do that immediately rather than wait on a button push. 

Either use the breadcrumbs that can be made present to use or get rid of them so that no one can use them. (and sure maybe leave behind commented out breadcrumbs for those that want to restore that functionality against recommendation.)

The other thing, yeah I think the wording is not quite right. Resume a session that has timed out? Depending on the background of the reader, could be (mis-)interpreted any number of ways. From the computer side it appears that the session would pick up right from where it left off with all portions of the session resumed (not at all what is intended), from the other spectrum likely just to mean that able to restore one's settings. 

It seems that although edits may be lost when the session period has ended, but on page load the current uri is/was available to be applied against the "end-of-session" script such that "continuing" would be possible at least from the same page and like the same product. If the expectation is that the user is to be returned to the beginning, then why wait and not just return them to the login screen without the added "taunting"? :)

**Comment by scottcwilson** [comment]
This issue should be closed.  The application is working as expected.
